tests/test_single_topology.py

```diff
@@ -2005,3 +2005,42 @@ def test_chiral_bond_breaking_1_core_1_dummy(seed):
 
     # should not raise an assertion
     verify_chiral_validity_of_core(mol_a, mol_b, core, ff)
+
+
+def get_vacuum_system_and_conf(mol_a, mol_b, core, lamb):
+    ff = Forcefield.load_default()
+    st = SingleTopology(mol_a, mol_b, core, ff)
+    conf_a = get_romol_conf(mol_a)
+    conf_b = get_romol_conf(mol_b)
+    conf = st.combine_confs(conf_a, conf_b, lamb)
+    return st.setup_intermediate_state(lamb), conf
+
+
+@pytest.mark.nocuda
+@pytest.mark.nightly(reason="slow")
+def test_hif2a_end_state_symmetry():
+    """
+    Test that end-states are symmetric
+    """
+    with resources.path("timemachine.testsystems.data", "ligands_40.sdf") as path_to_ligand:
+        mols = read_sdf(path_to_ligand)
+
+    for idx, mol_a in enumerate(mols):
+        for mol_b in mols[idx + 1 :]:
+            print("testing", mol_a.GetProp("_Name"), "->", mol_b.GetProp("_Name"))
+            core = atom_mapping.get_cores(mol_a, mol_b, **DEFAULT_ATOM_MAPPING_KWARGS)[0]
+            sys_fwd, conf_fwd = get_vacuum_system_and_conf(mol_a, mol_b, core, 0.0)
+            sys_rev, conf_rev = get_vacuum_system_and_conf(mol_b, mol_a, core[:, ::-1], 1.0)
+
+            box = 100.0 * np.eye(3)
+
+            assert sys_fwd.chiral_atom
+            assert sys_rev.chiral_atom
+            assert sys_fwd.torsion
+            assert sys_rev.torsion
+
+            np.testing.assert_allclose(sys_fwd.bond(conf_fwd, box), sys_rev.bond(conf_rev, box))
+            np.testing.assert_allclose(sys_fwd.angle(conf_fwd, box), sys_rev.angle(conf_rev, box))
+            np.testing.assert_allclose(sys_fwd.nonbonded(conf_fwd, box), sys_rev.nonbonded(conf_rev, box))
+            np.testing.assert_allclose(sys_fwd.chiral_atom(conf_fwd, box), sys_rev.chiral_atom(conf_rev, box))
+            np.testing.assert_allclose(sys_fwd.torsion(conf_fwd, box), sys_rev.torsion(conf_rev, box))
```

Comment:
Opportunity to make this more sensitive by asserting on forces as well, along the lines of 

![image](https://github.com/user-attachments/assets/18008478-39f5-4752-a261-33edae2ced22)

(for the pair `(338, 43)` @mcwitt had identified)

```python
import matplotlib.pyplot as plt

term_dict = {
  'bonds': (sys_fwd.bond, sys_rev.bond),
  'angles': (sys_fwd.angle, sys_rev.angle),
  'torsions': (sys_fwd.torsion, sys_rev.torsion),
  'chiral_atom': (sys_fwd.chiral_atom, sys_rev.chiral_atom),
  'nonbonded': (sys_fwd.nonbonded, sys_rev.nonbonded),
}

plt.figure(figsize=(15,4))
ax = None
for i, term in enumerate(term_dict):
  U_fwd, U_rev = term_dict[term]
  dudx_fwd = grad(U_fwd)(conf_fwd, box)
  dudx_rev = grad(U_rev)(conf_rev, box)
  ax = plt.subplot(1,5,i+1, sharey=ax)
  plt.title(term)
  plt.plot(np.linalg.norm(dudx_fwd[core[:,0]] - dudx_rev[core[:,1]], axis=1), '.')
  plt.yscale('log')
  plt.xlabel('idx within `core`')
  if i == 0:
    plt.ylabel('norm(dudx_fwd[core[idx,0]] - dudx_rev[core[idx,1]])')

plt.tight_layout()
```

Reply:
Thanks for the great suggestion. I modified the logic in 34f5126 to also include the dummy atoms as well.